modules/bibdk_recommender/js/bibdk_recommender.js

```diff
@@ -7,26 +7,28 @@
 // @codingStandardsIgnoreStart
 (function($) {
 
-  // Attach behaviour
+  // Attach Slick Recommender.
   Drupal.behaviors.initRecommenderExample = {
     attach: function(context) {
-      $( document, context).ready(function() {
-        $('.js-slick-recommender', context).once('slick-recommender').each(function(index, element) {
-          Drupal.getSlickRecommender($(this));
-        });
+      $('.js-slick-recommender', context).once('slick-recommender').each(function(index, element) {
+        Drupal.getSlickRecommender($(this));
       });
     }
   };
 
   // Retrieve Slick Recommender carousel
   Drupal.getSlickRecommender = function(elem) {
-    id = elem.attr("id");
-    pids = elem.attr("data-uids");
+    var id = elem.attr("id");
+    var pids = elem.attr("data-recomole-pids");
```

Comment:
It's generally recommended to use `.data` for data attributes, as it also supports data being stored in jQuery and no just reading the actual html attributes.
What happens if element is set but is empty? Shouldn't you check for non empty value rather than an undefined value?